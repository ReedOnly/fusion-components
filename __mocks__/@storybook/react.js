import renderer from 'react-test-renderer';
import addons, { mockChannel } from '@storybook/addons';
import { withFusionContext } from '../../.storybook/withFusionContext';

addons.setChannel(mockChannel());

export const action = (actionName) => jest.fn();
const fusionContext = withFusionContext();
export const storiesOf = (groupName) => {
    const api = {
        decorators: [],
        parameters: {},

        add(storyName, story) {
            if (api.parameters.skipTest) {
                return;
            }

            describe(groupName, () => {
                it(storyName, () => {
                    // const decoratedStory = api.decorators.reduce((s, decorator) => {
                    //     try {
                    //         return decorator(s, { parameters: api.parameters });
                    //     } catch (e) {
                    //         return s;
                    //     }
                    // }, story);

                    const component = renderer.create(fusionContext(story));

                    // @TODO Enable custom names once released:
                    // > https://github.com/facebook/jest/pull/2094
                    expect(component.toJSON())
                        .toMatchSnapshot
                        // `${groupName}.${storyName}`
                        ();
                });
            });

            return api;
        },

        addDecorator() {
            return api;
        },

        addParameters(parameters) {
            api.parameters = { ...api.parameters, ...parameters };
            return api;
        },
    };

    return api;
};

export const addDecorator = () => {};
