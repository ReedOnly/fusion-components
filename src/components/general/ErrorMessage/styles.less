@import "~styles/variables.less";

.container {
    position: relative;
    width: 100%;
    height: 100%;

    .messageContainer {
        position: absolute;
        top: 0; left: 0; bottom: 0; right: 0;
        display: flex;
        align-content: center;
        align-items: center;
        justify-content: center;
        flex-direction: column;
        background: rgba(255, 255, 255, .8);
        z-index: 999;

        .message {
            font-size: 14px;
            line-break: 24px;
            margin-bottom: calc(var(--grid-unit) * 2px);
            text-align: center;
            p{
                font-weight: 500;
            }
        }
        .icon {
            font-size: 100px;
            color: #cccccc;
        }
    }
}