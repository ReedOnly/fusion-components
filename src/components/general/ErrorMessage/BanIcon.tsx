import * as React from "react";

export default ({ scale=1 }) => (
    <svg width={72} height={72} viewBox="0 0 72 72" fill="none" xmlns="http://www.w3.org/2000/svg" style={{ transform: `scale(${scale})` }}>
        <path fillRule="evenodd" clipRule="evenodd" d="M69 36C69 54.2254 54.2256 69 36 69C27.6885 69 20.0938 65.9271 14.292 60.8555L60.7129 14.1297C65.8691 19.9523 69 27.6105 69 36ZM58.624 11.9758C52.7168 6.41058 44.7568 3 36 3C17.7744 3 3 17.7746 3 36C3 44.8343 6.47168 52.8578 12.125 58.7809L58.624 11.9758ZM72 36C72 55.8823 55.8818 72 36 72C16.1182 72 0 55.8823 0 36C0 16.1177 16.1182 0 36 0C55.8818 0 72 16.1177 72 36Z" fill="#CCCCCC"/>
    </svg>
);