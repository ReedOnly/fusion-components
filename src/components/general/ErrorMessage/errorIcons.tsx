import * as React from "react";

export const AccessDenied = (
    <svg width="80" height="80" viewBox="0 0 80 80" fill="none" xmlns="http://www.w3.org/2000/svg" preserveAspectRatio="none">
        <path
            fillRule="evenodd"
            clipRule="evenodd"
            d="M39.9994 6.66663C21.5993 6.66663 6.66602 21.6 6.66602 40C6.66602 58.4 21.5993 73.3333 39.9994 73.3333C58.3994 73.3333 73.3327 58.4 73.3327 40C73.3327 21.6 58.3994 6.66663 39.9994 6.66663ZM13.3327 40C13.3327 25.2666 25.266 13.3333 39.9994 13.3333C46.166 13.3333 51.8327 15.4333 56.3327 18.9666L18.966 56.3333C15.4327 51.8333 13.3327 46.1666 13.3327 40ZM23.666 61.0333C28.166 64.5666 33.8327 66.6666 39.9993 66.6666C54.7327 66.6666 66.666 54.7333 66.666 40C66.666 33.8333 64.566 28.1666 61.0327 23.6666L23.666 61.0333Z"
            fill="#666666"
        />
    </svg>
);

export const ErrorOrNotFound = (
    <svg width="80" height="80" viewBox="0 0 80 80" fill="none" xmlns="http://www.w3.org/2000/svg">
        <path
            fillRule="evenodd"
            clipRule="evenodd"
            d="M14.8998 68.3457C9.76642 68.3457 6.56642 62.779 9.13309 58.3457L34.2331 14.9791C36.7997 10.5457 43.1998 10.5457 45.7664 14.9791L70.8664 58.3457C73.4331 62.779 70.2331 68.3457 65.0997 68.3457H14.8998ZM65.0998 61.6791L39.9998 18.3124L14.8998 61.6791H65.0998ZM36.6664 35.0124V41.6791C36.6664 43.5124 38.1664 45.0124 39.9998 45.0124C41.8331 45.0124 43.3331 43.5124 43.3331 41.6791V35.0124C43.3331 33.1791 41.8331 31.6791 39.9998 31.6791C38.1664 31.6791 36.6664 33.1791 36.6664 35.0124ZM43.3331 58.3457V51.6791H36.6664V58.3457H43.3331Z"
            fill="#666666"
        />
    </svg>
);

export const noData = (
    <svg width="80" height="80" viewBox="0 0 80 80" fill="none" xmlns="http://www.w3.org/2000/svg">
        <path
            fillRule="evenodd"
            clipRule="evenodd"
            d="M67.8706 38.4458C67.8706 31.0792 64.8372 24.4458 60.0039 19.6458L65.0372 14.6125C66.0706 13.5792 65.3372 11.7792 63.8372 11.7792H47.8706V27.7458C47.8706 29.2458 49.6706 29.9792 50.7039 28.9125L55.3372 24.2792C58.9372 27.9458 61.2039 32.9125 61.2039 38.4458C61.2039 41.2792 60.6039 43.9792 59.5372 46.4125L64.4706 51.3458C66.6039 47.5125 67.8706 43.1458 67.8706 38.4458ZM34.5372 17.5792V16.9458C34.5372 14.6792 32.1706 13.2458 30.1372 14.2125C29.715 14.4125 29.3224 14.6421 28.9298 14.8718C28.7335 14.9866 28.5372 15.1014 28.3372 15.2125L33.2039 20.0792C34.0039 19.4792 34.5372 18.5792 34.5372 17.5792ZM13.1039 18.8125C11.8039 17.5125 11.8039 15.4125 13.1039 14.1125C13.7267 13.4883 14.5722 13.1376 15.4539 13.1376C16.3356 13.1376 17.1811 13.4883 17.8039 14.1125L66.2039 62.5458C67.5039 63.8458 67.5039 65.9458 66.2039 67.2458C64.9039 68.5458 62.8039 68.5458 61.5039 67.2458L55.2706 61.0125C54.3039 61.6125 53.3039 62.1792 52.2706 62.6458C50.2372 63.6125 47.8706 62.1792 47.8706 59.9125V59.3125C47.8706 57.8792 48.9372 56.6792 50.3706 56.2792C50.3706 56.2476 50.3706 56.231 50.3741 56.2152C50.3781 56.1976 50.3864 56.1809 50.4039 56.1458L23.5039 29.2458C22.0706 32.0125 21.2039 35.1125 21.2039 38.4458C21.2039 43.9792 23.4706 48.9458 27.0706 52.6125L31.7039 47.9792C32.7372 46.9125 34.5372 47.6458 34.5372 49.1458V63.4458C34.5372 64.3792 33.8039 65.1125 32.8706 65.1125H18.5706C17.0706 65.1125 16.3372 63.3125 17.3706 62.2792L22.4039 57.2458C17.5706 52.4458 14.5372 45.8125 14.5372 38.4458C14.5372 33.2458 16.0372 28.4458 18.6039 24.3125L13.1039 18.8125Z"
            fill="#666666"
        />
    </svg>
);