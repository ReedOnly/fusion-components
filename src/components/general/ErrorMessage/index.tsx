import * as React from 'react';
import Button from '../Button';
import styles from './styles.less';
import classNames from 'classnames';
import { useComponentDisplayClassNames } from '@equinor/fusion';
import { BlockIcon } from 'components/icons/components/content';
import WarningIcon from 'components/icons/components/alert/WarningIcon';
import SyncDisabledIcon from 'components/icons/components/notification/SyncDisabledIcon';

export type ErrorTypes = 'error' | 'accessDenied' | 'notFound' | 'noData';

export type ErrorMessageProps = {
    hasError?: boolean;
    errorType?: ErrorTypes;
    message?: any;
    resourceName?: string;
    title?: string;
    children?: any;
    icon?: any;
    action?: string;
    onTakeAction?: (event?: React.SyntheticEvent<Element, Event>) => void;
};

const ErrorMessage: React.FC<ErrorMessageProps> = ({
    hasError,
    errorType = 'error',
    message,
    resourceName,
    title,
    children,
    icon,
    action,
    onTakeAction,
}) => {
    const getErrorMessageForType = errorType => {
        const iconProps = {
            width: 80,
            height: 80,
            cursor: 'default',
            color: '#666666',
        };

        switch (errorType) {
            case 'accessDenied':
                return {
                    title: 'It seems like you donÂ´t have access to this content',
                    icon: <BlockIcon {...iconProps} />,
                };
            case 'notFound':
                return {
                    title: `The ${resourceName} could not be found`,
                    icon: <WarningIcon {...iconProps} outline />,
                };
            case 'noData':
                return {
                    title: `Unfortunately, we could not find any data for this component`,
                    icon: <SyncDisabledIcon {...iconProps} />,
                };
            default:
                return {
                    title: 'Oops! Something went wrong!',
                    icon: <WarningIcon {...iconProps} outline />,
                };
        }
    };

    if (!hasError) {
        return children;
    }
    const error = React.useMemo(() => getErrorMessageForType(errorType), [errorType]);

    const messageContainerClasses = classNames(
        styles.messageContainer,
        useComponentDisplayClassNames(styles)
    );

    return (
        <div className={styles.container}>
            <div className={messageContainerClasses}>
                {icon || error.icon}
                <div className={styles.title}>{title || error.title}</div>
                <div className={styles.message}>{message}</div>
                {action ? (
                    <Button outlined contained onClick={onTakeAction}>
                        {action}
                    </Button>
                ) : null}
            </div>
        </div>
    );
};

export default ErrorMessage;
