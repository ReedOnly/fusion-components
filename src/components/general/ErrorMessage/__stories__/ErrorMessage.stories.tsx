import * as React from "react";
import { storiesOf } from "@storybook/react";
import ErrorMessage from "../index";

const ErrorMessageStory = () =>{
    
    return(
        <ErrorMessage message="This the message for the Error Message" hasError/>
    )
}

storiesOf("General components/ErrorMessage", module)
    .addParameters({ jest: ["ErrorMessage.stories.tsx"] })
    .add("ErrorMessage", () => <ErrorMessageStory />);
