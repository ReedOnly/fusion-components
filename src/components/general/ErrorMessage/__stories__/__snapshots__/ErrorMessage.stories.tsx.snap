// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General|ErrorMessage Error 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <div
      className="container"
    >
      <div
        className="messageContainer comfortable"
      >
        <svg
          cursor="default"
          height={80}
          style={
            Object {
              "color": "#666666",
            }
          }
          viewBox="0 0 24 24"
          width={80}
        >
          <path
            clipRule="evenodd"
            d="M4.47 20.5038C2.93 20.5038 1.97 18.8338 2.74 17.5038L10.27 4.49378C11.04 3.16378 12.96 3.16378 13.73 4.49378L21.26 17.5038C22.03 18.8338 21.07 20.5038 19.53 20.5038H4.47ZM19.53 18.5038L12 5.49378L4.47 18.5038H19.53ZM11 10.5038V12.5038C11 13.0538 11.45 13.5038 12 13.5038C12.55 13.5038 13 13.0538 13 12.5038V10.5038C13 9.95378 12.55 9.50378 12 9.50378C11.45 9.50378 11 9.95378 11 10.5038ZM13 17.5038V15.5038H11V17.5038H13Z"
            fill="currentColor"
            fillRule="evenodd"
          />
        </svg>
        <div
          className="title"
        >
          Oops! Something went wrong!
        </div>
        <div
          className="message"
        >
          Your error message describing how to resolve the problem
        </div>
        <button
          className="container contained outlined primary comfortable"
          disabled={false}
          onClick={[Function]}
          onClickCapture={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          type="button"
        >
          <span
            className="button"
          >
            Resolve the problem
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;

exports[`General|ErrorMessage NoAccess 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <div
      className="container"
    >
      <div
        className="messageContainer comfortable"
      >
        <svg
          cursor="default"
          height={80}
          style={
            Object {
              "color": "#666666",
            }
          }
          viewBox="0 0 24 24"
          width={80}
        >
          <path
            clipRule="evenodd"
            d="M10 0C4.48 0 0 4.48 0 10C0 15.52 4.48 20 10 20C15.52 20 20 15.52 20 10C20 4.48 15.52 0 10 0ZM2 10C2 5.58 5.58 2 10 2C11.85 2 13.55 2.63 14.9 3.69L3.69 14.9C2.63 13.55 2 11.85 2 10ZM5.1 16.31C6.45 17.37 8.15 18 10 18C14.42 18 18 14.42 18 10C18 8.15 17.37 6.45 16.31 5.1L5.1 16.31Z"
            fill="currentColor"
            fillRule="evenodd"
          />
        </svg>
        <div
          className="title"
        >
          It seems like you donÂ´t have access to this content
        </div>
        <div
          className="message"
        >
          Your error message describing how to resolve the problem
        </div>
        <button
          className="container contained outlined primary comfortable"
          disabled={false}
          onClick={[Function]}
          onClickCapture={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          type="button"
        >
          <span
            className="button"
          >
            Resolve the problem
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;

exports[`General|ErrorMessage NoData 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <div
      className="container"
    >
      <div
        className="messageContainer comfortable"
      >
        <svg
          cursor="default"
          height={80}
          style={
            Object {
              "color": "#666666",
            }
          }
          viewBox="0 0 24 24"
          width={80}
        >
          <path
            clipRule="evenodd"
            d="M17.3613 8.53375C17.3613 6.32375 16.4513 4.33375 15.0013 2.89375L16.5113 1.38375C16.8213 1.07375 16.6013 0.533752 16.1513 0.533752H11.3613V5.32375C11.3613 5.77375 11.9013 5.99375 12.2113 5.67375L13.6013 4.28375C14.6813 5.38375 15.3613 6.87375 15.3613 8.53375C15.3613 9.38375 15.1813 10.1938 14.8613 10.9238L16.3413 12.4038C16.9813 11.2538 17.3613 9.94375 17.3613 8.53375ZM7.36129 2.27375V2.08375C7.36129 1.40375 6.65129 0.973752 6.04129 1.26375C5.91463 1.32375 5.79685 1.39264 5.67907 1.46153C5.62018 1.49598 5.56129 1.53042 5.50129 1.56375L6.96129 3.02375C7.20129 2.84375 7.36129 2.57375 7.36129 2.27375ZM0.931294 2.64375C0.541294 2.25375 0.541294 1.62375 0.931294 1.23375C1.11813 1.0465 1.37178 0.941267 1.63629 0.941267C1.90081 0.941267 2.15446 1.0465 2.34129 1.23375L16.8613 15.7638C17.2513 16.1538 17.2513 16.7838 16.8613 17.1738C16.4713 17.5638 15.8413 17.5638 15.4513 17.1738L13.5813 15.3038C13.2913 15.4838 12.9913 15.6538 12.6813 15.7938C12.0713 16.0838 11.3613 15.6538 11.3613 14.9738V14.7938C11.3613 14.3638 11.6813 14.0038 12.1113 13.8838C12.1113 13.8743 12.1113 13.8693 12.1124 13.8645C12.1135 13.8593 12.116 13.8543 12.1213 13.8438L4.05129 5.77375C3.62129 6.60375 3.36129 7.53375 3.36129 8.53375C3.36129 10.1938 4.04129 11.6838 5.12129 12.7838L6.51129 11.3938C6.82129 11.0738 7.36129 11.2938 7.36129 11.7438V16.0338C7.36129 16.3138 7.14129 16.5338 6.86129 16.5338H2.57129C2.12129 16.5338 1.90129 15.9938 2.21129 15.6838L3.72129 14.1738C2.27129 12.7338 1.36129 10.7438 1.36129 8.53375C1.36129 6.97375 1.81129 5.53375 2.58129 4.29375L0.931294 2.64375Z"
            fill="currentColor"
            fillRule="evenodd"
          />
        </svg>
        <div
          className="title"
        >
          Unfortunately, we could not find any data for this component
        </div>
        <div
          className="message"
        >
          Your error message describing how to resolve the problem
        </div>
        <button
          className="container contained outlined primary comfortable"
          disabled={false}
          onClick={[Function]}
          onClickCapture={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          type="button"
        >
          <span
            className="button"
          >
            Resolve the problem
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;

exports[`General|ErrorMessage NotFound 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <div
      className="container"
    >
      <div
        className="messageContainer comfortable"
      >
        <svg
          cursor="default"
          height={80}
          style={
            Object {
              "color": "#666666",
            }
          }
          viewBox="0 0 24 24"
          width={80}
        >
          <path
            clipRule="evenodd"
            d="M4.47 20.5038C2.93 20.5038 1.97 18.8338 2.74 17.5038L10.27 4.49378C11.04 3.16378 12.96 3.16378 13.73 4.49378L21.26 17.5038C22.03 18.8338 21.07 20.5038 19.53 20.5038H4.47ZM19.53 18.5038L12 5.49378L4.47 18.5038H19.53ZM11 10.5038V12.5038C11 13.0538 11.45 13.5038 12 13.5038C12.55 13.5038 13 13.0538 13 12.5038V10.5038C13 9.95378 12.55 9.50378 12 9.50378C11.45 9.50378 11 9.95378 11 10.5038ZM13 17.5038V15.5038H11V17.5038H13Z"
            fill="currentColor"
            fillRule="evenodd"
          />
        </svg>
        <div
          className="title"
        >
          The person could not be found
        </div>
        <div
          className="message"
        >
          Your error message describing how to resolve the problem
        </div>
      </div>
    </div>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;
